### PR TITLE
Updates for Swift 3

### PR DESCRIPTION
**Pull request by bensyverson**
Here are the changes for Swift 3—heads up that the Package.swift is pointing to my BinaryJSON repo

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -5,6 +5,6 @@ let package = Package(
     dependencies: [
         .Package(url: "https://github.com/PureSwift/CBSON.git", majorVersion: 1),
         .Package(url: "https://github.com/Danappelxx/CMongoC.git", majorVersion: 0, minor: 1),
-        .Package(url: "https://github.com/Danappelxx/BinaryJSON", majorVersion: 1, minor: 1)
+        .Package(url: "https://github.com/bensyverson/BinaryJSON", majorVersion: 1, minor: 5)
```
Comment (Danappelxx) [inline Package.swift]: Could you undo this please? :stuck_out_tongue:

File: Sources/Client.swift [hunk 2]
```diff
@@ -54,9 +54,9 @@ public final class Client {
 
         var names = [String]()
 
-        while buffer.memory != nil {
+        while buffer.pointee != nil {
 
-            let name = String.fromCString(buffer.memory)!
+			let name = String(cString:buffer.pointee)
```
Comment (Danappelxx) [inline Sources/Client.swift]: whitespace

File: Sources/Collection.swift [hunk 3]
```diff
@@ -27,7 +27,7 @@ public final class Collection {
         mongoc_collection_destroy(self.pointer)
     }
 
-    public func insert(document: BSON.Document, flags: InsertFlags = .None) throws {
+    public func insert(document document: BSON.Document, flags: InsertFlags = .None) throws {
```
Comment (Danappelxx) [inline Sources/Collection.swift]: Does this break anything? I would imagine it does.

File: Sources/Database.swift [hunk 4]
```diff
@@ -24,7 +24,7 @@ public final class Database {
 
     public var name: String {
         let nameRaw = mongoc_database_get_name(pointer)
-        return String.fromCString(nameRaw)!
+		return String(cString:nameRaw)
```
Comment (Danappelxx) [inline Sources/Database.swift]: whitespace

File: Sources/Database.swift [hunk 5]
```diff
@@ -38,9 +38,9 @@ public final class Database {
 
         var names = [String]()
 
-        while buffer.memory != nil {
+        while buffer.pointee != nil {
 
-            let name = String.fromCString(buffer.memory)!
+			let name = String(cString:buffer.pointee)
```
Comment (Danappelxx) [inline Sources/Database.swift]: whitespace

File: Sources/MongoError.swift [hunk 6]
```diff
@@ -74,9 +74,9 @@ public extension bson_error_t {
     }
 }
 
-func mongocErrorMessageToString(inout error: _mongoc_error_message) -> String {
+func mongocErrorMessageToString(error: inout _mongoc_error_message) -> String {
     return withUnsafePointer(&error) {
-        String.fromCString(UnsafePointer($0))!
+		String(cString:UnsafePointer($0))
```
Comment (Danappelxx) [inline Sources/MongoError.swift]: whitespace

File: Sources/Collection.swift (hunk 3 above)
Comment (bensyverson) [inline Sources/Collection.swift]: I don't think I added that—should I remove it?

File: Package.swift (hunk 1 above)
Comment (bensyverson) [inline Package.swift]: Sure! What minor version should I use?

File: Sources/Collection.swift (hunk 3 above)
Comment (Danappelxx) [inline Sources/Collection.swift]: Yeah, its causing the tests to fail so please do :)

Comment (bensyverson) [inline Sources/Collection.swift]: Ok, removed—so weird, I don't know where it came from. Let me know what minor version to use for BinaryJSON, and I'll push an update.

Comment (Danappelxx) [inline Sources/Collection.swift]: I'm waiting for the ci at BinaryJSON to pass (I just configured it for 03-24) before merging & tagging a release, but the version is going to be `1.5`

Comment (bensyverson) [inline Sources/Collection.swift]: Word! Just pushed those changes. :smile:

**Comment by Danappelxx** [comment]
CI isn't working here either, but again - I don't see a reason why anything could break. Thanks for the pr!

**Comment by bensyverson** [comment]
Awesome, glad to help! I just saw the Open Swift stuff by the way—that looks really great. Hopefully all of these web frameworks will stop reinventing the wheel and share some common standards. Thanks for your work on SwiftMongoDB—this is going to come in handy!

**Comment by Danappelxx** [comment]
Yeah! That's been our goal with Zewo from the start but OpenSwift makes our intentions more clear :smiley:

**Pull request merged**